src/lib/registration/registration_codes_loader.rb

```diff
@@ -0,0 +1,100 @@
+# encoding: utf-8
+
+# ------------------------------------------------------------------------------
+# Copyright (c) 2016 SUSE, LLC. All Rights Reserved.
+#
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, contact Novell, Inc.
+#
+# To contact Novell about this file by physical or electronic mail, you may find
+# current contact information at www.novell.com.
+# ------------------------------------------------------------------------------
+
+require "tempfile"
+
+require "yast"
+require "transfer/file_from_url"
+
+Yast.import "XML"
+
+module Registration
+  # Aids Registration::Storage::RegCodes in loading the codes
+  # from external storage so that the user does not have to type them
+  # each time when doing multiple installations. FATE#316796
+  module RegCodesLoader
```

Comment:
NTH: I think it would be nice to have some unit tests for the `RegCodesLoader` module with some testing data files in `test/fixtures`...